server/studyspaces.py

```diff
@@ -1,54 +1,24 @@
 from flask import request, jsonify
 from server import app
-from bs4 import BeautifulSoup
-import requests
+from .penndata import *
 
 
-def get_id_json():
-    """
-    Extracts the ID's of the room to a JSON.
-    """
-    groupStudyCodes = []
-    url = "http://libcal.library.upenn.edu/booking/vpdlc"
-    soup = BeautifulSoup(requests.get(url).text, 'lxml')
-    l = soup.find_all('option')
-    for element in l:
-        if element['value'] != '0':
-            url2 = "http://libcal.library.upenn.edu" + str(element['value'])
-            soup2 = BeautifulSoup(requests.get(url2), 'lxml')
-            id = soup2.find('input', attrs={"id": "gid"})['value']
-            newDict = {}
-            newDict['id'] = int(id)
-            newDict['name'] = element.contents[0]
-            groupStudyCodes.append(newDict)
-    return groupStudyCodes
-
-
-def get_id_dict():
-    """
-    Extracts the ID's of the room into a dictionary.
-    """
-    groupStudyCodes = {}
-    url = "http://libcal.library.upenn.edu/booking/vpdlc"
-    soup = BeautifulSoup(requests.get(url).text, 'lxml')
-    l = soup.find_all('option')
-    for element in l:
-        if element['value'] != '0':
-            url2 = "http://libcal.library.upenn.edu" + str(element['value'])
-            soup2 = BeautifulSoup(requests.get(url2).text, 'lxml')
-            id = soup2.find('input', attrs={"id": "gid"})['value']
-            groupStudyCodes[int(id)] = element.contents[0]
-    return groupStudyCodes
-
-
-@app.route('/studyspaces/<date>', methods=['GET'])
+@app.route('/studyspaces/avail/<date>', methods=['GET'])
 def parse_times(date):
     """
     Returns JSON with available rooms.
+
+    Usage:
+        /studyspaces/avail/<date> gives all available
+        /studyspaces/avail/<date>?id=<id> gives the avaiable room with <id>
     """
     d = get_id_dict()
     if 'id' in request.args:
-        id = request.args['id']
+        try:
+            id = request.args['id']
+        except KeyError:
+            # check for invalid ID's
+            return jsonify({'studyspaces': "Invalid ID number."})
```

Comment:
I also don't like this JSON object, doesn't make much sense to me. `{'error': "Invalid ID number."}` makes much more sense to me, some variant of that works too.